api/v1beta1/placementapi_webhook.go

```diff
@@ -121,18 +126,27 @@ func (r *PlacementAPI) ValidateDelete() (admission.Warnings, error) {
 }
 
 func (r PlacementAPISpec) ValidateCreate(basePath *field.Path) field.ErrorList {
-	return r.ValidateDefaultConfigOverwrite(basePath)
+	return ValidateDefaultConfigOverwrite(basePath, r.DefaultConfigOverwrite)
 }
 
 func (r PlacementAPISpec) ValidateUpdate(old PlacementAPISpec, basePath *field.Path) field.ErrorList {
-	return r.ValidateDefaultConfigOverwrite(basePath)
+	return ValidateDefaultConfigOverwrite(basePath, r.DefaultConfigOverwrite)
+}
+
+func (r PlacementAPISpecCore) ValidateCreate(basePath *field.Path) field.ErrorList {
+	return ValidateDefaultConfigOverwrite(basePath, r.DefaultConfigOverwrite)
+}
+
+func (r PlacementAPISpecCore) ValidateUpdate(old PlacementAPISpec, basePath *field.Path) field.ErrorList {
+	return ValidateDefaultConfigOverwrite(basePath, r.DefaultConfigOverwrite)
 }
 
-func (r PlacementAPISpec) ValidateDefaultConfigOverwrite(
+func ValidateDefaultConfigOverwrite(
```

Comment:
An alternative would be to change this to use the PlacementAPISpecCore as a receiver. Then 
* `func (r PlacementAPISpecCore) ValidateCreate` can call it directly as the type matches
* `func (r PlacementAPISpec) ValidateCreate` can call it on its internal core struct within its receiver. Golang actually expose the functions on the embedded struct on the embedding struct automatically.

Reply:
This would reduce the change to https://github.com/gibizer/placement-operator/commit/91e9a074e1300722060710acdda4bc90c349cd82

While this is not a big deal here in placement, I think if we apply this pattern to the nova PR that would reduce the change there significantly

Reply:
Applying it to nova is not a clear cut due to deeper nesting of structs without a common embedded base struct.